src/PKSim.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -26,9 +27,18 @@ public ProjectMapper(IExecutionContext executionContext)
          snapshot.Formulations = await mapBuildingBlocksToSnapshots<Formulation>(project.All<Model.Formulation>());
          snapshot.Protocols = await mapBuildingBlocksToSnapshots<Protocol>(project.All<Model.Protocol>());
          snapshot.Populations = await mapBuildingBlocksToSnapshots<Population>(project.All<Model.Population>());
+         snapshot.ObservedData = await mapObservedDataToSnapshots(project.AllObservedData);
+
          return snapshot;
       }
 
+      private async Task<DataRepository[]> mapObservedDataToSnapshots(IReadOnlyCollection<OSPSuite.Core.Domain.Data.DataRepository> allObservedData)
+      {
+         var snapshotMapper = _executionContext.Resolve<ObservedDataMapper>();
+         var tasks = allObservedData.Select(datarepository => snapshotMapper.MapToSnapshot(datarepository));
+         return await Task.WhenAll(tasks);
```

Comment:
see above. This is the correct way to do it
That being said, you do not need the async and await here again